### ISSUE TITLE
Add multiplayer

### ISSUE THREAD

**Issue by ChrisNeedham24** [ChrisNeedham24/microcosm#117]
Eventually, it would be cool to have multiplayer functionality. Probably will be P2P.

**Comment by ChrisNeedham24** [comment]
Could also make it client-server, probably would be easier, and a basic Digital Ocean droplet is only $4 a month.

**Comment by ChrisNeedham24** [comment]
Probably best to do this once the core game functionality is somewhat settled so that changes don't affect this.

**Comment by ChrisNeedham24** [comment]
While core functionality is basically done now, I think this is something I'd probably want to do post-#20

**Comment by ChrisNeedham24** [comment]
The more I think about it, the more I realise that #20 will likely take a very long time to implement - I'll do some investigation on this issue to see if it could be any quicker. If so, this might be the first cab off the rank.

**Comment by ChrisNeedham24** [comment]
At this point, my investigation borders on actually doing it, so I'll commit.

**Comment by ChrisNeedham24** [comment]
Will want to add some more statistics and achievements for this too.

**Comment by ChrisNeedham24** [comment]
When this is done it would be worth making a new issue to update to Python 3.12 - there's a chunking method in `itertools` now that could have been useful.

**Comment by ChrisNeedham24** [comment]
A number of files will probably be over 1000 lines at the end of this - should make new issues to refactor those once this is done.

**Comment by ChrisNeedham24** [comment]
Will need to update the screenshot when done

**Comment by ChrisNeedham24** [comment]
Might be worth making an issue once this is done to convert the save file format to use the new minified quads format

**Comment by ChrisNeedham24** [comment]
Also will need to update the README to mention routers needing UPnP

**Comment by ChrisNeedham24** [comment]
Once this is done, make an issue to implement turn timers (and potentially other game settings)

**Comment by ChrisNeedham24** [comment]
Will need to migrate old float yields for quads to be ints - just round

**Comment by ChrisNeedham24** [comment]
Once this is done, make a new issue to expand save naming - include faction, turn, game name for multiplayer? Basically just make it easier to tell which save is the one you're looking for.

**Comment by ChrisNeedham24** [comment]
Probably will want to do a sweep over the entire codebase just to make sure everything still makes sense.

**Comment by ChrisNeedham24** [comment]
Make an issue to account for desync - hash maybe?
